src/App.tsx

```diff
@@ -1,14 +1,105 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+const debounce = (f: (...args: string[]) => void, delay: number) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(() => {
+      f(...args);
+    }, delay);
+  };
+};
+
+const DEBOUNCE_DELAY = 1000;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Todo[]>([]);
+  const [sortMethod, setSortMethod] = useState('all');
+  const [applieQuery, setApplieQuery] = useState('');
+  const [iLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+  const [visibiliteComponent, setVisibiliteComponent] = useState(false);
+
+  const apllyQuery = useCallback(
+    debounce(setApplieQuery, DEBOUNCE_DELAY),
+    [],
+  );
+
+  const handleFilterChange = (methodSort: string) => {
+    setSortMethod(methodSort);
+  };
+
+  const handleQueryText = (queryText: string) => {
+    apllyQuery(queryText);
+  };
+
+  useEffect(() => {
+    const newFilterTodos = todos.filter(todo => {
+      switch (sortMethod) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    }).filter(todo => {
+      return todo.title.toLowerCase().includes(applieQuery.toLowerCase());
+    });
+
+    setFilterTodos(newFilterTodos);
+  }, [todos, sortMethod, applieQuery]);
```

Comment:
Do not use useEffect and additional state for this case - create `const` that will have `visibleTodos`.
Note no need to filter `todos` in case when `sortMethod` is all and `query` is empty